### PR TITLE
Ensure that tunneled payloads are parsed

### PR DESCRIPTION
**Pull request by hardillb**
## Description

HTTP Body is JSON.stringify() before being sent down the tunnel need to JSON.parse() before forwarding on to editor

## Related Issue(s)

flowfuse/flowfuse#2999

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
For reference here is the other end of the tunnel `JSON.stringify()` call

**Comment by knolleary** [comment]
With this change, I can load the device editor, but I cannot deploy any changes

**Comment by hardillb** [comment]
When you get a second can you look at this revised fix, shouldn't break anything this time and still allow UIBuilder to work.

**Comment by hardillb** [comment]
@Steve-Mcl If you get a moment can you kick the tires on this please. I've given it a better test than the first version and I'm 99% sure it won't break anything this time.

File: lib/editor/tunnel.js [hunk 1]
```diff
@@ -169,10 +169,20 @@ class EditorTunnel {
...
+                    // debug(`body ${request.body}`)
+                    let body = request.body
+                    try {
+                        const temp = JSON.parse(body)
+                        if (typeof temp === 'string') {
+                            body = temp
+                        }
+                    } catch (err) {
+                        // ignore
+                    }
```
Comment (knolleary) [inline lib/editor/tunnel.js:181]:
```suggestion
                    // debug(`body ${request.body}`)
                    let body = request.body
                    if (body && body[0] === '"') {
                        try {
                            body = JSON.parse(body)
                        } catch (err) {
                            // ignore
                        }
                    }
```

This feels an expensive way to only parse strings. Suggest adding a check to see if it looks like a JSON encoded String before attempting the parse.

**Review by knolleary — CHANGES_REQUESTED** [review]

**Review by knolleary — APPROVED** [review]

**Pull request merged**